core/src/test/scala/shapeless/poly.scala

```diff
@@ -370,4 +370,37 @@ class PolyTests {
     val r6 = dcis(2.0, 'a', 1, "foo")
     assertTypedEquals[String](s"i: 1, s: foo, d: ${2.0}, c: a", r6)
   }
+
+  @Test
+  def testPoly1Builder {
+    val myPoly = Poly1.at[Int]( x => x).at[String](_.length).at[Boolean](if(_) 1 else 0).build
+    import myPoly._
+
+    val r1 = myPoly(10)
+    assertTypedEquals[Int](10, r1)
+
+    val r2 = myPoly("hello")
+    assertTypedEquals[Int](5, r2)
+
+    val r3 = myPoly(true)
+    assertTypedEquals[Int](1, r3)
+  }
+
+  @Test
+  def testPoly2Builder {
+    val myPoly = Poly2.at[Int, Int]((acc, x) => acc + x).
+                                at[Int, String]((acc, s) => acc + s.length).
+                                at[Int, Boolean]((acc, b) => acc + (if(b) 1 else 0)).
+                                build
+    import myPoly._
```

Comment:
Whitespace looks odd here ... stray tabs? Could you either line up the `at`'s or shallow indent in two space increments.